__tests__/integration.js

```diff
@@ -151,6 +154,40 @@ test('--mutex network', async () => {
   await Promise.all(promises);
 });
 
+test('--mutex network with busy port', async () => {
+  const port = getRandomPort();
+
+  const server = http.createServer((request, response) => {
+    response.writeHead(200);
+    response.end("I'm a broken JSON string to crash Yarn network mutex.");
+  });
+  server.listen({
+    port,
+    host: 'localhost',
+  });
+
+  const cwd = await makeTemp();
+  await fs.writeFile(
+    path.join(cwd, 'package.json'),
+    JSON.stringify({
+      scripts: {test: 'node -e "setTimeout(function(){}, process.argv[1])"'},
+    }),
+  );
+
+  let mutexError;
+  try {
+    await runYarn(['--mutex', `network:${port}`, 'run', 'test', '100'], {cwd});
+  } catch (error) {
+    mutexError = error;
+  } finally {
+    server.close();
+  }
+
+  expect(mutexError).toBeDefined();
```

Comment:
I guess this line is now obsolete with the `invariant` statement below? (added invariant later to silence flow)

Reply:
Yeah but I'd still keep it to make it explicit that we want to test it.